src/sleep/sleep.c

```diff
@@ -316,10 +313,10 @@ static int custom_timer_suspend(const SleepConfig *sleep_config) {
                 woken_by_timer = FLAGS_SET(r, POLLIN);
 
                 r = fetch_batteries_capacity_by_name(&current_capacity);
-                if (r < 0) {
+                if (r < 0 || hashmap_isempty(current_capacity)) {
                         /* In case of no battery or error while getting charge level, no need to measure
-                         * discharge rate. Instead system should wakeup if it is manual wakeup or
-                         * hibernate if this is a timer wakeup.   */
+                         * discharge rate. Instead system should wakeup if it is manual wakeup or hibernate
+                         * if this is a timer wakeup. */
                         log_debug_errno(r, "Battery capacity percentage unavailable, cannot estimate discharge rate: %m");
```

Comment:
if `r` is actually not < 0 this will hit an assert...

Reply:
I don't see an assert around here?

Reply:
`log_*_errno()` asserts that the error argument is negative.
(We shouldn't print "Battery capacity percentage unavailable, cannot estimate discharge rate: Success".)

The case of `hashmap_isempty(current_capacity)` needs a different log invocation.

Reply:
ah, forgot about that, thanks